theta/theta_sketch.cpp

```diff
@@ -150,96 +99,35 @@ EMSCRIPTEN_BINDINGS(theta_sketch) {
+      intersection.update(wrapped_compact_theta_sketch::wrap(bytes2.data(), bytes2.size(), seed));
+      const auto bytes = intersection.get_result().serialize_compressed();
+      return Uint8Array.new_(emscripten::typed_memory_view(bytes.size(), bytes.data()));
     }))
     ;
 
-  emscripten::class_<theta_a_not_b>("theta_a_not_b")
-    .constructor(emscripten::optional_override([](uint64_t seed) {
-      return new theta_a_not_b(seed);
-    }))
-    .function("computeWithCompactSketch", emscripten::optional_override([](theta_a_not_b& self,
-      const compact_theta_sketch& sketch1, const compact_theta_sketch& sketch2) {
-      return self.compute(sketch1, sketch2);
-    }))
-    .function("computeWithWrappedSketch", emscripten::optional_override([](theta_a_not_b& self,
-      const wrapped_compact_theta_sketch& sketch1, const wrapped_compact_theta_sketch& sketch2) {
-      return self.compute(sketch1, sketch2);
-    }))
-    .function("computeWithB64ReturnB64", emscripten::optional_override([](theta_a_not_b& self,
-      const std::string& b64_1, const std::string& b64_2, uint64_t seed) {
-      std::vector<char> bytes1(b64_dec_len(b64_1.data(), b64_1.size()));
-      b64_decode(b64_1.data(), b64_1.size(), bytes1.data());
-      std::vector<char> bytes2(b64_dec_len(b64_2.data(), b64_2.size()));
-      b64_decode(b64_2.data(), b64_2.size(), bytes2.data());
-      auto bytes = self.compute(
-        wrapped_compact_theta_sketch::wrap(bytes1.data(), bytes1.size(), seed),
-        wrapped_compact_theta_sketch::wrap(bytes2.data(), bytes2.size(), seed)
-      ).serialize();
-      std::vector<char> b64(b64_enc_len(bytes.size()));
-      b64_encode((const char*) bytes.data(), bytes.size(), b64.data());
-      return std::string(b64.data(), b64.size());
-    }))
-    .function("computeWithB64ReturnB64Compressed", emscripten::optional_override([](theta_a_not_b& self,
-      const std::string& b64_1, const std::string& b64_2, uint64_t seed) {
-      std::vector<char> bytes1(b64_dec_len(b64_1.data(), b64_1.size()));
-      b64_decode(b64_1.data(), b64_1.size(), bytes1.data());
-      std::vector<char> bytes2(b64_dec_len(b64_2.data(), b64_2.size()));
-      b64_decode(b64_2.data(), b64_2.size(), bytes2.data());
-      auto bytes = self.compute(
+  emscripten::function("thetaAnotBCompressed", emscripten::optional_override([](const std::string& bytes1, const std::string& bytes2, uint64_t seed) {
+      auto a_not_b = theta_a_not_b(seed);
+      const auto bytes = a_not_b.compute(
         wrapped_compact_theta_sketch::wrap(bytes1.data(), bytes1.size(), seed),
         wrapped_compact_theta_sketch::wrap(bytes2.data(), bytes2.size(), seed)
       ).serialize_compressed();
-      std::vector<char> b64(b64_enc_len(bytes.size()));
-      b64_encode((const char*) bytes.data(), bytes.size(), b64.data());
-      return std::string(b64.data(), b64.size());
+      return Uint8Array.new_(emscripten::typed_memory_view(bytes.size(), bytes.data()));
```

Comment:
should we specify a return_value_policy on things like this? (it's an actual question, not saying we need to)

Reply:
I am not sure